lib/error/lib/merr.c

```diff
@@ -49,7 +49,7 @@ merr_pack(int errno_value, const enum hse_err_ctx ctx, const char *file, int lin
 
     err |= ((uint64_t)line << MERR_LINE_SHIFT) & MERR_LINE_MASK;
     err |= (ctx << MERR_CTX_SHIFT) & MERR_CTX_MASK;
-    err |= errno_value & MERR_ERRNO_MASK;
+    err |= (ulong)errno_value & MERR_ERRNO_MASK;
```

Comment:
Isn't err a uint64_t ?

Reply:
err is a uint64_t, yes. Would you rather that be the cast?

Reply:
Yes please!